### PR TITLE
[12.0][FIX] sale_order_global_stock_route: Solved problems related with the method product_id_change()

### PR DESCRIPTION
**Pull request by CarlosRoca13**
When we add order lines and the data was updated using that method, the field of the lines was not updated. With this fix, we solve this and the module is easier to understand

cc @Tecnativa TT23962

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Well, this salomonical solution as product is mandatory on sales order lines works, although not very academic. The problem is if you create orders from code/XML-RPC. I think you need to override also sale.order.line create / write method for handling this. On v13 it will be easier thanks to computed writeable fields.

**Comment by pedrobaeza** [comment]
@sergio-teruel your opinion here is welcome.

**Comment by sergio-teruel** [comment]
Yes!! @CarlosRoca13 @pedrobaeza is right. By this way, any external application like as a mobile app  would have to call all onchange methods. But if we open this door... Do we have to apply this to all the modules that already exist? Anyway... You can extend write and create method instead of onchange method and you write a comment for remove this methods and use a computed writable fields....

**Comment by pedrobaeza** [comment]
Yeah, in v13 we fortunately can rid off this problem. @sergio-teruel if you don't consider that as needed for current use cases, then let's go as is.

File: sale_order_global_stock_route/models/sale_order.py [hunk 1]
```diff
@@ -21,27 +20,27 @@ def _onchange_route_id(self):
...
+    _inherit = 'sale.order.line'
+
+    @api.onchange('product_id')
+    def product_id_change(self):
+        res = super().product_id_change()
+        self.route_id = self.order_id.route_id
+        return res
+
+    @api.model
+    def create(self, vals_list):
```
Comment (pedrobaeza) [inline sale_order_global_stock_route/models/sale_order.py:42]: Make use of create multi here

**Review by pedrobaeza — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**